share/functions/psub.fish

```diff
@@ -3,40 +3,41 @@ function psub --description "Read from stdin into a file and output the filename
 
 	set -l filename
 	set -l funcname
+	set -l suffix
 	set -l use_fifo 1
-	set -l shortopt -o hf
-	set -l longopt -l help,file
 
-	if getopt -T >/dev/null
-		set longopt
-	end
-
-	if not getopt -n psub -Q $shortopt $longopt -- $argv >/dev/null
-		return 1
-	end
-
-	set -l tmp (getopt $shortopt $longopt -- $argv)
-
-	eval set opt $tmp
-
-	while count $opt >/dev/null
+	while count $argv >/dev/null
 
-		switch $opt[1]
+		switch $argv[1]
 			case -h --help
 				__fish_print_help psub
 				return 0
 
 			case -f --file
 				set use_fifo 0
+				set -e argv[1]
+
+			case -s --suffix
+				set suffix $argv[2]
+				set -e argv[1..2]
 
 			case --
-				set -e opt[1]
+				set -e argv[1]
 				break
 
-		end
+			case "-?" "--*"
+				printf "psub: invalid option: '%s'\n" $argv[1]
+				return 1
 
-		set -e opt[1]
+			case "-*"
+				set opts (printf "-%s\n" (printf $argv[1] |grep -o "\w"))
```

Comment:
This can be done with our (new and awesome) `string` utility - something like `string sub -s 2 $argv[1] | string split ""`